vulture_os/applications/backend/views.py

```diff
@@ -353,11 +352,11 @@ def render_form(back, **kwargs):
             backend.save()
             logger.debug("Backend '{}' (id={}) saved in MongoDB.".format(backend.name, backend.id))
 
-            """ And all the listeners created earlier """
+            """ Create or update all the listeners """
             for s in server_objs:
                 s.backend = backend
-                logger.debug("Saving server {}".format(str(s)))
-                s.save()
+                logger.debug("Saving/updating server {}".format(str(s)))
+                Server.objects.update_or_create(target=s.target, port=s.port, defaults=s.to_dict())
```

Comment:
`s.to_dict()` won't work with the defaults, as serialized values format is not compatible